Cash in Advance assistant

kevinfink96 opened the pull request:
@plentymarkets/team-order-payment 
Kanbanize: https://plentymarkets.kanbanize.com/ctrl_board/75/cards/72926/details/

kevinfink96 committed:
assistant

kevinfink96 committed:
remove old settings ui and update plugin version

CTohme commented on src/Assistants/SettingsHandlers/CashInAdvanceSettingsHandler.php:
```diff
@@ -0,0 +1,216 @@
+        $dataListEntry['dataProviderPluginId'] = $prePaymentPlugin->id;
+        $dataListEntry['containerPluginId'] = $ceresPlugin->id;
+        $dataListEntry['pluginSetId'] = $webstore->pluginSetId;
+        $dataListEntry['dataProviderPluginSetEntryId'] = $prePaymentPlugin->pluginSetEntries[0]->id;
+        $dataListEntry['containerPluginSetEntryId'] = $ceresPlugin->pluginSetEntries[0]->id;
```
Warum nimmst du immer den ersten Eintrag? Kann es nicht auch mehrere geben?

CTohme commented on src/Services/SettingsService.php:
```diff
@@ -358,6 +398,30 @@ private function loadClientSettings($plentyId, $lang)
+        {
+            $query->where('lang', '=', $lang);
+        }
+        $query->orWhere('lang',   '=', '')->where('plentyId', '=', $plentyId);
```
Warum fügst du die Where-Bedingung: `->orWhere('lang',   '=', '')` immer hinzu, selbst in dem Fall falls eine Sprache vorhanden ist...?

kevinfink96 replied:
Es gibt manche Optionen, die ohne eine Sprache gespeichert werden.
![Bildschirmfoto 2019-10-02 um 14 11 43](https://user-images.githubusercontent.com/50198131/66043151-b0d46580-e51e-11e9-833d-343a210183d5.png)

CTohme reviewed (commented):
Die Änderungen sehen soweit i.O. aus. Zwei Fragen zu deinem Code hatte ich noch. 

- Testen konnte ich es leider noch nicht, weil ich im Augenblick noch Probleme mit dem Payment-Assistenten in meinem lokalen System habe.

kevinfink96 committed:
logo fix

kevinfink96 committed:
changes from invoice assistant

kevinfink96 committed:
add lang variable to settings array

kevinfink96 committed:
moved payment description to CMS -> multilingualism

kevinfink96 committed:
update changelog, user guide and naming for the assistent

kevinfink96 committed:
added missing container connections

kevinfink96 committed:
save settings global instead of only for one language

kevinfink96 committed:
moved payment method name to CMS -> multilingualism

kevinfink96 committed:
update changelog date

kevinfink96 committed:
changelog wording

kevinfink96 committed:
don't select logo in footer as default

j-konopka commented on meta/documents/user_guide_de.md:
```diff
@@ -7,163 +7,10 @@ Mit diesem Plugin binden Sie die Zahlungsart **Vorkasse** in Ihren Webshop ein.
-
-Der Kunde kann nun die Zahlungsart in der Auftragshistorie im **Mein-Konto**-Bereich bzw. auf der Bestellbestätigungsseite neu ausführen. Bei Vorkasse bedeutet dies, dass die im Backend hinterlegten Bankinformationen erneut angezeigt werden.
+1. Öffnen Sie das Menü **Einrichtung&nbsp;» Assistenten**.
+2. Wählen Sie den Assistenten **Vorkasse**.
```
Hier muss noch der richtige Bereich "Payment" hin

j-konopka commented on meta/documents/user_guide_en.md:
```diff
@@ -8,163 +8,10 @@ In order to make this payment method available in your online store, you have to
-
-The customer can now reinitialise the payment either in the order history in the **My Account** area or on the order confirmation page. For Cash in advance, this means accessing once more the bank data saved in the backend.
+1. Go to **Setup&nbsp;» Assistants**.
+2. Select the assistant **Cash in advance**.
```
Gleiches wie im deutschen. Der Bereich "payment" fehlt hier.

j-konopka commented on src/Assistants/CashInAdvanceAssistant.php:
```diff
@@ -0,0 +1,342 @@
+                ];
+            }
+
+            usort($this->webstoreValues, function ($a, $b) {
```
War die Sortierung eine Anforderung?

j-konopka commented on src/Assistants/CashInAdvanceAssistant.php:
```diff
@@ -0,0 +1,342 @@
+                ];
+            }
+            // Sort values alphabetically
+            usort($this->deliveryCountries, function($a, $b) {
```
War die Sortierung eine Anforderung?

kevinfink96 replied:
Nein, habe die Sortierung jetzt entfernt.
Die Funktion hatte ich vom Lastschrift-Assistenten übernommen.

j-konopka commented on src/Assistants/DataSources/AssistantDataSource.php:
```diff
@@ -0,0 +1,169 @@
+                $data[$pid]['config_name'] = $pid;
+                $data[$pid]['logo_url'] = $data[$pid]['logoUrl'];
+                $data[$pid]['logo_type_external'] = $data[$pid]['logo'] == 1;
+                $data[$pid]['PaymentMethodIcon'] = $this->logoInFooter($pid);
```
Hieß der Parameter schon immer so?
Schreibweise ist ja unterschiedlich zu den anderen...

kevinfink96 committed:
update changelog date
added payment path to the user guide
removed the sorting of county and webstore
renamed PaymentMethodIcon

kevinfink96 committed:
update changelog: setup path
